assets/app/features/upgrade/controllers/upgrade-upgrade-administration-server.controller.js

```diff
@@ -48,21 +52,25 @@
                 .then(
                     // In case of success
                     function (response) {
-                        // TODO: update to match API
-                        vm.adminUpgrade.completed = response.data.completed;
+                        // map api response to model
+                        vm.adminUpgrade.completed = response.data.upgrade.success;
+                        vm.adminUpgrade.running = response.data.upgrade.upgrading;
                     },
                     // In case of failure
                     function (errorResponse) {
                         // Expose the error list to adminUpgrade object
+                        // TODO(itxaka): Use the proper error key from the response
                         vm.adminUpgrade.errors = errorResponse.data.errors;
                     }
                 ).finally(
                     function () {
                         // schedule another check if not completed yet
-                        if (!vm.adminUpgrade.completed) {
+                        if (!vm.adminUpgrade.completed && vm.adminUpgrade.running) {
```

Comment:
This is done to avoid the timeout from firing on page load, as we are now calling the checkAdminUpgrade on controller creation, it wiould always trigger it. Instead we check both values so it only triggers it when the upgrade is ongoing.